src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+                             "|"                                     // or MMDDCCYY
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]{4})"
+                             ")",
+
+                             "(?:"                                   // either d-m(-y)
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<MONTH>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or DDMM(CCYY)
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+
+                             "(?:"                                   // either m-d(-y)
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<DAY>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or MMDD(CCYY)
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+};
+//const int num_currency_formats = 3;
+//const gchar* currency_format_user[] = {N_("Locale"),
+//                                       N_("Period: 123,456.78"),
+//                                       N_("Comma: 123.456,78")
+//                                      };
+//
+/* This map contains a set of strings representing the different column types. */
+std::map<GncTransPropType, const char*> gnc_csv_col_type_strs = {
+        { GncTransPropType::NONE, N_("None") },
+        { GncTransPropType::DATE, N_("Date") },
+        { GncTransPropType::NUM, N_("Num") },
+        { GncTransPropType::DESCRIPTION, N_("Description") },
+        { GncTransPropType::NOTES, N_("Notes") },
+        { GncTransPropType::ACCOUNT, N_("Account") },
+        { GncTransPropType::DEPOSIT, N_("Deposit") },
+        { GncTransPropType::WITHDRAWAL, N_("Withdrawal") },
+        { GncTransPropType::BALANCE, N_("Balance") },
+        { GncTransPropType::MEMO, N_("Memo") },
+        { GncTransPropType::OACCOUNT, N_("Other Account") },
+        { GncTransPropType::OMEMO, N_("Other Memo") }
+};
+
+/** Parses a string into a date, given a format. This function
+ * requires only knowing the order in which the year, month and day
+ * appear. For example, 01-02-2003 will be parsed the same way as
+ * 01/02/2003.
+ * @param date_str The string containing a date being parsed
+ * @param format An index specifying a format in date_format_user
+ * @return The parsed value of date_str on success or -1 on failure
+ */
+time64 parse_date (const std::string &date_str, int format)
```

Comment:
At present it expects ISO formatted strings, so there are two ways to go:

1. Add a constructor ```GncDateTime(const std::string& date_str, const std::string& format_str)``` where ```format_str``` is a ```strftime()```-style format, e.g. "%m-%d-%Y".
2. Have ```parse_date``` just reorder the string (and add the current year if necessary) to an unambiguous ISO date, YYYY-MM-DD and pass that to ```GncDateTime(const std::string&)```.

I'm inclined towards 1.